### PR TITLE
Add build script to download language server if not found

### LINKED ISSUE #3107: Gracefully recover from failure to detect the language server project in sibling directory
Our [`CONTRIBUTING.md` server-side setup](https://github.com/redhat-developer/vscode-java/blob/master/CONTRIBUTING.md#b-server-side-setup) talks about how to set up vscode-java with eclipse.jdt.ls (placing them as sibling directories) in order to ensure `npm run build-server` builds eclipse.jdt.ls and uses the resulting content in the `server/` folder of vscode-java.

However, we also have a gulp task that fetches the latest pre-built language server (eclipse.jdt.ls) from http://download.eclipse.org/jdtls/snapshots/jdt-language-server-latest.tar.gz . It would be good to integrate this workflow in the case a user has not properly set up a sibling directory.

I don't know if I'd do this for `npm run build-server` though maybe we create a separate task called `npm run build` that either performs `build-server` if eclipse.jdt.ls is detected, or the download_server task.

### PR DESCRIPTION
**Pull request by hopehadfield**
Fixes #3107

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -81,8 +81,9 @@ This assumes that you are starting on the `vscode-java` directory
 
 	```bash
 	$ cd ./vscode-java
-	$ npm run build-server
+	$ npm run build
 	```
+If the eclipse.jdt.ls directory is not found as a sibling directory (see the desired hierarchy in the previous step), a snapshot of the language server will be downloaded instead. You will see a note to indicate this incident.
```
Comment (rgrunber) [inline CONTRIBUTING.md:86]: I would remove the "You will see a note to indicate this incident.". If it happens, the log message will be clear.

**Review by rgrunber — CHANGES_REQUESTED** [review]
Overall, it works pretty well! Just some minor improvements that can be made.

**Review by JessicaJHee — COMMENTED** [review]
Tested and works great! Just left some very minor suggestions

File: gulpfile.js [hunk 2]
```diff
@@ -161,6 +161,23 @@ gulp.task('build_server', function (done) {
...
+			.pipe(decompress())
+			.pipe(gulp.dest('./server'));
+	}
+	else {
+		cp.execSync(mvnw() + ' -Pserver-distro clean package -Declipse.jdt.ls.skipGradleChecksums', { cwd: server_dir, stdio: [0, 1, 2] });
+		gulp.src(server_dir + '/org.eclipse.jdt.ls.product/distro/*.tar.gz')
+			.pipe(decompress())
+			.pipe(gulp.dest('./server'));
+	}
+	done();
```
[the remark this answers is not in the document]
Comment (hopehadfield) [reply gulpfile.js:178]: Just looked into it and there's something called gulp.start() to initiate another task from within a task, but it's deprecated. I can create functions for the logic in download_server and build_server instead and just call the function from each task if that works for you?

[the remark this answers is not in the document]
Comment (rgrunber) [reply gulpfile.js:178]: Yes! That's pretty much the same conclusion I came to.  Needed to create a function at the bottom and then just have the gulp tasks call it, followed by `done()`.

**Pull request merged**